src/git.rs

```diff
@@ -97,13 +97,18 @@ impl GitWrapper for Git {
         self.run_cmd(remote_update_cmd)
     }
 
-    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf) -> Result {
+    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf, refspec: &Option<String>) -> Result {
         let mut push_cmd = self.git_base_cmd();
-        push_cmd
-            .current_dir(repo_dir)
-            .args(&["push", "--mirror"])
-            .arg(dest);
-
+        push_cmd.current_dir(repo_dir);
+        push_cmd.arg("push");
+        match &refspec {
```

Comment:
[`if let`](https://doc.rust-lang.org/book/ch06-03-if-let.html) would be more readable here.